datajoint/external.py

```diff
@@ -192,8 +178,32 @@ def clean_store(self, store, verbose=True):
                         os.rmdir(folder)
                         deleted_folders.add(folder)
             print('Deleted %d objects' % next(count))
-        elif spec['protocol'] == 's3':
+        elif self.spec['protocol'] == 's3':
             try:
-                failed_deletes = s3.Folder(database=self.database, **spec).clean(in_use, verbose=verbose)
+                failed_deletes = s3.Folder(database=self.database, **self.spec).clean(in_use, verbose=verbose)
             except TypeError:
-                raise DataJointError('External store {store} configuration is incomplete.'.format(store=store))
+                raise DataJointError('External store {store} configuration is incomplete.'.format(store=self.store))
+
+
+class ExternalMapping:
+    """
+    The external manager contains all the tables for all external stores for a given schema
+    :Example:
+        e = ExternalMapping(schema)
+        external_table = e[store]
+    """
+    def __init__(self, schema):
+        self.schema = schema
+        self.external_tables = {}
+
+    def __getitem__(self, store):
+        if store not in self.external_tables:
+            self.external_tables[store] = ExternalTable(
+                connection=self.schema.connection, store=store, database=self.schema.database)
+        return self.external_tables[store]
+
+    def __iter__(self):
```

Comment:
can we get a `keys()` too?
andor alternately could external mapping generally not be done via some more complete and well thought out notion of `DefaultDict(lambda k: ExternalTable(config[store]))` ?

Reply:
actually, thinking more not sure if i like the 'simple' dict approach since custom class could allow things like maintenance methods, like 'consistency check', 'bulk cleanup' etc to happen at 'schema' level.

Reply:
getting keys still useful - unless i'm mistaken there is not a clear/simple way to retrieve 'names' of stores w/o knowing internals of the mapping since `list(schema.external)[0].spec` has config 'values' but not 'names'.. but could be overlooking something.

Reply:
Yes, I am thinking that `ExternalMapping` should derive from `collections.Mapping` and act like a dict with additional methods.

Reply:
This has been fixed now. `schema.external`is a Mapping.